src/scripts/generate-test-patient-spreadsheets.sh

```diff
@@ -0,0 +1,186 @@
+
+  log "INFO" "Requesting ${url}"
+  if [ "200" != "$(curl -s -o "${curlResponse}" -w "%{http_code}" -H "Authorization: Bearer $TOKEN" "${url}")" ]; then
+    log "INFO" "Refreshing token and retrying ${url}"
+    TOKEN=$(new-token)
+    statusCode=$(curl -s -o "${curlResponse}" -w "%{http_code}" -H "Authorization: Bearer $TOKEN" "${url}")
+    if [ "200" != "${statusCode}" ]; then
+      log "ERROR" "Request to ${url} failed. Status code was ${statusCode}."
+      exit 1
+    fi
+  fi
+}
+
+generateCsv() {
+  local outputCsv="${1}"
+  local curlResponse baseUrl patientFields resourceStaticFields resourceJqFields patientFieldValues staticFieldValues
+  curlResponse=$(mktemp -p "${WORK:-/tmp}")
+  baseUrl=$(jq -r '.baseUrl' "${TEMPLATE_FILE}")
+
+  patientFields=$(jq -r '.patientFields[].name' "${TEMPLATE_FILE}" | tr '\n' ' ')
+  resourceStaticFields=()
+  resourceJqFields=()
+
+  for i in $(seq 0 $(($(jq '.resources | length' "${TEMPLATE_FILE}")-1))); do
+    for staticField in $(jq -r ".resources[${i}].staticFields[].name" "${TEMPLATE_FILE}"); do
+      if [[ ! " ${resourceStaticFields[*]} " =~ [[:space:]]${staticField}[[:space:]] ]]; then
+          resourceStaticFields+=" ${staticField}"
+      fi
+    done
+    for jqField in $(jq -r ".resources[${i}].jqFields[].name" "${TEMPLATE_FILE}"); do
+      if [[ ! " ${resourceJqFields[*]} " =~ [[:space:]]${jqField}[[:space:]] ]]; then
+          resourceJqFields+=" ${jqField}"
+      fi
+    done
+  done
+
+  echo "ICN,${patientFields// /,}Resource${resourceStaticFields// /,}${resourceJqFields// /,}" > "${outputCsv}"
+
+
+  TOKEN=$(new-token)
+
+  # Send the requests and create the spreadsheet rows
+  for patientId in $(jq -r '.testPatients[]' "${TEMPLATE_FILE}"); do
+    request "${baseUrl}/Patient/${patientId}" "${curlResponse}"
+    patientFieldValues="${patientId}"
+    for field in ${patientFields}; do
+      patientFieldValues+=","
+      value=$(jq -r "$(jq -r ".patientFields[] | select(.name==\"${field}\") | .path" "${TEMPLATE_FILE}")" "${curlResponse}")
+      if [ "${value}" != "null" ]; then
+        patientFieldValues+="${value//,/ }"  # Replace commas with spaces
+      fi
+    done
+
+    for i in $(seq 0 $(($(jq '.resources | length' "${TEMPLATE_FILE}")-1))); do
+      resourceType=$(jq -r ".resources[${i}].type" "${TEMPLATE_FILE}")
+      staticFieldValues=""
+      for field in ${resourceStaticFields}; do
+        staticFieldValues+=","
+        value="$(jq -r ".resources[] | select( .type==\"${resourceType}\") | ( .staticFields // [])[] | select(.name==\"${field}\") | .value" "${TEMPLATE_FILE}")"
+        if [ "${value}" != "null" ]; then
+          staticFieldValues+="${value//,/ }"  # Replace commas with spaces
+        fi
+      done
+      createSpreadsheetRowsForResource "${resourceType}" "${outputCsv}" "${curlResponse}" "${patientId}" "${patientFieldValues},${resourceType}${staticFieldValues}" "${resourceJqFields}"
+    done
+  done
```

Comment:
I wonder if theres a way to parallelize this by resource. How long does it take to run this on average and generate the whole spreadsheet?